src/main/java/com/intuit/graphql/orchestrator/schema/transform/FieldResolverTransformerPostMerge.java

```diff
@@ -53,6 +69,24 @@ public XtextGraph transform(XtextGraph sourceXtextGraph) {
     return sourceXtextGraph;
   }
 
+  private void validateRequiredFields(FieldResolverContext fieldResolverContext) {
```

Comment:
Possibly refactor and move the validation to its own class(FieldResolveContext) so the transformer performs one task. Also performs less peek operations making it cleaner

Reply:
Yup agree.  Good suggestion.   Will refactor

Reply:
this is done